installbuilder/datafiles/Base_DSC.data

```diff
@@ -169,14 +169,16 @@ EOF
 
 chmod u+x $tmpfile
 $tmpfile 1> /dev/null 2> /dev/null
-if [ $? -ne 0 ]; then
+tmpfile_ctypes=$?
+rm $tmpfile
+
+python -c "import ctypes" 1> /dev/null 2> /dev/null
```

Comment:
The problem is: There's nothing in the script form that ties us to `/tmp`. This would mean, I strongly suspect, that Python itself opens a temporary file (on `/tmp`), assuming you're running a script. And, if so, that would be pretty unfortunate.

Given that there are likely further problems (with Python itself) if `/tmp` is mounted noexec, I'd rather (if you want to keep both checks) give a more specific error message. After all, if Python has problems running scripts because of `/tmp`, it would be way better to give the customer a specific error message (mentioning `/tmp` perhaps being noexec).

Could you change the code to try both forms and, if one fails, give a better error message? Probably check inline form first, then check script form second, and fail if either fails.